samvel-aivazian/lesson10-hw4.py

```diff
@@ -0,0 +1,48 @@
+# Movie database
+movie_database = {
+    "Interstellar": ["Adventure", "Drama", "Sci-Fi"],
+    "Inception": ["Action", "Adventure", "Sci-Fi"],
+    "The Dark Knight": ["Action", "Crime", "Drama"],
+    "Pulp Fiction": ["Crime", "Drama"],
+    "The Matrix": ["Action", "Sci-Fi"],
+    "The Godfather": ["Crime", "Drama"],
+    "Shawshank Redemption": ["Drama"],
+    "The Avengers": ["Action", "Adventure", "Sci-Fi"],
+    "Titanic": ["Drama", "Romance"],
+    "Finding Nemo": ["Animation", "Adventure", "Comedy"]
+}
+
+
+def get_user_preferences() -> list:
+    genres = input("Enter your preferred genres (comma separated): ")
+    preferred_genres = [genre.strip() for genre in genres.split(',')]
+    return preferred_genres
+
+
+def recommend_movie(preferred_genres: list) -> str:
+    recommendations = []
+    for movie, genres in movie_database.items():
+        if any(genre in genres for genre in preferred_genres):
+            recommendations.append(movie)
+
+    if recommendations:
+        return recommendations
+    else:
+        return "No movies found matching your preferences."
```

Comment:
There is no need to add an additional `else` clause

Could be:

```python
    if recommendations:
        return recommendations
    return "No movies found matching your preferences."
```

The typing is wrong as you return not only `list` but `str` as well. I't's a bad approach, in terms of maintainability.

Reply:
> The typing is wrong as you return not only list but str as well

Don't get me wrong, sometimes it's unavoidable, but in this case it's an easy fix